gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/ManagedHttpJsonChannel.java

```diff
@@ -150,8 +183,20 @@ public Builder setHttpTransport(HttpTransport httpTransport) {
     public ManagedHttpJsonChannel build() {
       Preconditions.checkNotNull(endpoint);
 
+      // If the executor provided for this channel is null, gax will provide a
+      // default executor to used for the calls. Only the default executor's
+      // lifecycle will be managed by the channel. Any external executor needs to
+      // managed by the user.
+      boolean usingDefaultExecutor = executor == null;
+      if (usingDefaultExecutor) {
+        executor = InstantiatingExecutorProvider.newBuilder().build().getExecutor();
+      }
```

Comment:
The following is more readable to me, we need to add `usingDefaultExecutor` as a field to the builder as well, which is also a best practice for using builders.
```suggestion
      if (executor == null) {
        executor = InstantiatingExecutorProvider.newBuilder().build().getExecutor();
        usingDefaultExecutor = false;
      }
```

Reply:
Sure, will update! I had a bit of hesitation with adding a setter for `usingDefaultExecutor` since I was afraid people would do some combination of setting:
- `usingDefaultExecutor` = false && `executor` = null
- `usingDefaultExecutor` = true && `executor` = non-null value
which are kinda invalid states.

Then `build()` would override the `usingDefaultExecutor` value anyways. Probably not best practice for builder, but I think `usingDefaultExecutor` should be internal anyways.

Reply:
We can add `usingDefaultExecutor` as a private field but not have to expose it through a setter.

Reply:
Ah, I misunderstood. I can add a field for it.